File: go/solutions/2894-divisible-and-non-divisible-sums-difference.go
package solutions

func DifferenceOfSums(n int, m int) int {
	divSum, nonDivSum := 0, 0
	for i := 1; i <= n; i++ {
		if i%m == 0 {
			divSum += i
		} else {
			nonDivSum += i
		}
	}

	return nonDivSum - divSum
}

File: js/2894-divisible-and-non-divisible-sums-difference.js
/**
 * @param {number} n
 * @param {number} m
 * @return {number}
 */
var differenceOfSums = function (n, m) {
  var divSum = 0,
    nonDivSum = 0;
  for (var i = 1; i <= n; i++) {
    if (i % m == 0) {
      divSum += i;
    } else {
      nonDivSum += i;
    }
  }

  return nonDivSum - divSum;
};

module.exports = { differenceOfSums };
